### PR TITLE
WIP: Allow serialization of selected trajectory IDs.

### PR DESCRIPTION
**Pull request by MichaelGrupp**
The optional parameter `trajectory_ids` allows to serialize only
a subset of the data. This is for example useful for re-mapping
on top of an existing map and serializing only the new map.

### CODE REVIEW

**Comment by MichaelGrupp** [comment]
Follow-up for `cartographer_ros`: https://github.com/googlecartographer/cartographer_ros/pull/749

**Comment by MichaelGrupp** [comment]
@cschuet This also affects the grpc interface, I just added the optional param there without functionality yet. What's your opinion?

**Comment by MichaelGrupp** [comment]
CLA bump

File: cartographer/mapping/map_builder.cc [hunk 1]
```diff
@@ -168,7 +168,10 @@ std::string MapBuilder::SubmapToProto(
   return "";
 }
 
-void MapBuilder::SerializeState(io::ProtoStreamWriterInterface* const writer) {
+void MapBuilder::SerializeState(io::ProtoStreamWriterInterface* const writer,
+                                const std::set<int> trajectory_ids) {
```
Comment (hauke76) [inline cartographer/mapping/map_builder.cc]: Should this by any chance be a `const std::set<int>&` ?

Comment (MichaelGrupp) [reply cartographer/mapping/map_builder.cc]: Yes. The params will change anyway as I used a default parameter value in the `map_builder_interface.h` in this prototype, which is not conform to the Google style guide.

**Comment by MichaelGrupp** [comment]
As discussed in the open house session, this is more a simple prototype to illustrate the behavior. Some interface details like the function params need to be discussed.

If a more thorough discussion regarding this feature is needed (e.g. more selection features than just trajectory IDs), I'm also fine with opening an RFC. @gaschler

**Comment by MichaelGrupp** [comment]
Closing in favor of WIP on new trimmers.

**Pull request closed without merging**